src/sentry/integrations/gitlab/repository.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import absolute_import
+
+import six
+
+from sentry.plugins import providers
+from sentry.models import Integration
+
+
+class GitlabRepositoryProvider(providers.IntegrationRepositoryProvider):
+    name = 'Gitlab'
+
+    def get_installation(self, integration_id, organization_id):
+        if integration_id is None:
+            raise ValueError('%s requires an integration_id' % self.name)
+
+        try:
+            integration_model = Integration.objects.get(id=integration_id)
+        except Integration.DoesNotExist as error:
+            self.handle_api_error(error)
+
+        return integration_model.get_installation(organization_id)
+
+    def validate_config(self, organization, config):
+        installation = self.get_installation(config['installation'], organization.id)
+        client = installation.get_client()
+
+        repo_id = config['identifier']
+        instance = installation.model.metadata['domain_name']
```

Comment:
Not necessary here, but this made me think of it.

We should make properties on the model for all of the attributes we _always_ expect to be in `metadata`. So we can `self.model.domain_name`, for example. Also might be nice to rename `model` to `integration` – it's pretty opaque currently. </random-thought>

Reply:
@mnoble so this is a bigger change that should probably involve @EvanPurkhiser who originally wrote this model. I don't have a good understanding as to where the dividing line should be between properties and json blobs. I agree with you since we always seem to have a domain_name and display it in the UI.

Reply:
Originally installation was called `integration`, so in that case `integration.model` made more sense.